asistente_ladm_col/tests/test_toolbar.py

```diff
@@ -0,0 +1,125 @@
+                                db_gpkg.names.LC_PLOT_T: 14,
+                                db_gpkg.names.MORE_BFS_T: 42,
+                                db_gpkg.names.LESS_BFS_T: 0,
+                                db_gpkg.names.POINT_BFS_T: 46}
+
+        for layer_name, feature_count in layers_feature_count.items():
+            self.assertEqual(layers[layer_name].featureCount(),
+                             feature_count,
+                             'Features count does not match for the layer {}'.format(layer_name))
+
+        print('\nINFO: Check build boundaries using a selection...')
+        boundary_ids = ['c1a08287-4fd5-4504-a6b7-cd029bd0d352', 'e7db709f-f9ef-4d83-b59b-692f1c4f7942',
+                        'db813999-3e1e-4539-99b6-7b797b1a3a72', '9831e428-040b-48e1-960a-5546fa47b99e',
+                        '9a009adf-6e87-4b04-aeb6-5d0830a848a7', '25509907-7cbc-4c7d-ab76-153898205f80',
+                        '1ef74b53-46fa-4dc5-9d85-c25db94f3dcd', '5b7194f2-54d8-4352-a47d-fac7f270a4a1']
+
+        exp = "{} in ('{}')".format(db_gpkg.names.T_ILI_TID_F,
+                                    "', '".join(boundary_ids))
+        layers[db_gpkg.names.LC_BOUNDARY_T].selectByExpression(exp)
+
+        self.toolbar.build_boundary(db_gpkg)
+
+        layers_feature_count = {db_gpkg.names.LC_BOUNDARY_POINT_T: 12,
+                                db_gpkg.names.LC_BOUNDARY_T: 22,
+                                db_gpkg.names.LC_PLOT_T: 14,
+                                db_gpkg.names.MORE_BFS_T: 43,
+                                db_gpkg.names.LESS_BFS_T: 0,
+                                db_gpkg.names.POINT_BFS_T: 47}
+
+        for layer_name, feature_count in layers_feature_count.items():
+            self.assertEqual(layers[layer_name].featureCount(),
+                             feature_count,
+                             'Features count does not match for the layer {}'.format(layer_name))
+
+        print('\nINFO: Check build boundaries for all features...')
+        layers[db_gpkg.names.LC_BOUNDARY_T].selectAll()
+        self.toolbar.build_boundary(db_gpkg)
+
+        layers_feature_count = {db_gpkg.names.LC_BOUNDARY_POINT_T: 12,
+                                db_gpkg.names.LC_BOUNDARY_T: 25,
+                                db_gpkg.names.LC_PLOT_T: 14,
+                                db_gpkg.names.MORE_BFS_T: 44,
+                                db_gpkg.names.LESS_BFS_T: 0,
+                                db_gpkg.names.POINT_BFS_T: 50}
+
+        for layer_name, feature_count in layers_feature_count.items():
+            self.assertEqual(layers[layer_name].featureCount(),
+                             feature_count,
+                             'Features count does not match for the layer {}'.format(layer_name))
+
+    def test_build_boundaries_with_empty_geom(self):
+        print('\nINFO: Validating build boundaries with features with null geometry...')
```

Comment:
Unclear what you're testing. 

What is the result you expect? The NULL geometry is ignored?

Reply:
the information given in the tests has been updated. This test validates that if there is a boundary without geometry it must be removed. vailable in commit 513f31a